rasa/core/events/__init__.py

```diff
@@ -680,39 +677,39 @@ class ReminderCancelled(Event):
 
     def __init__(
         self,
-        action_name: Text,
+        intent: Text,
         timestamp: Optional[int] = None,
         metadata: Optional[Dict[Text, Any]] = None,
     ):
         """
         Args:
-            action_name: name of the scheduled action to be cancelled
+            intent: name of the scheduled event to be cancelled
             metadata: optional event metadata
         """
 
-        self.action_name = action_name
+        self.intent = intent
         super().__init__(timestamp, metadata)
 
     def __hash__(self) -> int:
-        return hash(self.action_name)
+        return hash(self.intent)
 
     def __eq__(self, other) -> bool:
         return isinstance(other, ReminderCancelled)
 
     def __str__(self) -> Text:
-        return f"ReminderCancelled(action: {self.action_name})"
+        return f"ReminderCancelled(intent: {self.intent})"
 
     def as_story_string(self) -> Text:
-        props = json.dumps({"action": self.action_name})
+        props = json.dumps({"intent": self.intent})
```

Comment:
What about the entities here? (and everywhere else in reminder cancelled and reminder scheduled)

Reply:
I think `ReminderCancelled` should not be sensitive to entities, but only check for the intent name. This way, the developer does not have to worry about getting the "parameters" (entities) of the reminder she wants to cancel right, but only the kind of reminder.
I'll update the `ReminderScheduled` though - forgot about that.

Reply:
Ah, no. I actually didn't intent to put entities into the reminder. The reminder just triggers and intent, without entities. I could change that, though...

Reply:
Ok, I've implemented a support for entities for the `Reminder`, but not for `ReminderCancelled`, as mentioned above.

Reply:
That sounds decent to me. Do you know if it is possible to set 2 reminders at once? in that case I could see the use case of cancelling a specific one. The easiest theoretical example sth like saying set a timer for 30 mins -> `/set_timer{"minutes": 30}` -- if I set multiple timers, they are distinct based on the entity and I want to cancel a specific one.

Reply:
althogh i guess if it weren't possible you woudn't have to sent the intent anyway 😅

Reply:
Hmm... even when we'd compare entities, we could not distinguish two reminders that have the same intent and entities, but different tigger times.

Reply:
Another way would be to identify reminders by some sort of handle / id, that is distinct from the intent name and entitites. If that handle is not set, it is the intent name + channel by default, as it is now. What do you think?

Reply:
good point. I would move toward the default including the timestamp of when the reminder was set (not when it is scheduled to remind), to avoid duplicates

Reply:
I think some user may _want_ duplicates. E.g. when you want to kill all reminders that would trigger a certain intent, regardless of the parameters. So I'd leave it ambiguous (intents only) by default, but allow the user to have full control by assigning specific handles to each reminder.

Reply:
Ok, how about this: We add an ID field to Reminders that includes the creation time, as you suggest, but we make the `ReminderCancelled` event more flexible. In particular, the developer can send `ReminderCancelled` with any combination of parameters, depending on how specific she wants to be. For example:

* `ReminderCancelled(intent="greet")` cancels all reminders with intent `greet`

* `ReminderCancelled(entities={...})` cancels all reminders with the given entities

* `ReminderCancelled("2020-01-10...")` cancels the one unique reminder with the given id

* `ReminderCancelled()` cancels all reminders